src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -166,10 +171,79 @@ private boolean validateNucleusConfig(CompletableFuture<DeploymentResult> totall
                         .complete(new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE, e));
                 return false;
             }
+
+            Integer timeoutSec = deployment.getDeploymentDocumentObj().getConfigurationValidationPolicy().timeoutInSeconds();
+            long configTimeout = Duration.ofSeconds(20).toMillis();
+            if (timeoutSec != null) {
+                configTimeout = Duration.ofSeconds(timeoutSec).toMillis();
+            }
+
+            Map<String, Object> currentConfig = kernel.getConfig().toPOJO();
+            if (configTimeout == 0
+                    || !deviceConfiguration.isDeviceConfiguredToTalkToCloud()
+                    || !hasNucleusConfigurationChange(currentConfig, nucleusConfig)) {
+                logger.atInfo().log("Skipping connectivity validation");
+                return true;
+            }
+            try {
+                // Update Nucleus with new config
+                logger.atInfo().log("Applying Nucleus config");
+                Map<String, Object> modifiedConfig = kernel.getConfig().toPOJO();
+                replaceNucleusConfiguration(modifiedConfig, nucleusConfig);
+                activator.mergeConfig(modifiedConfig, deployment);
+                TimeUnit.SECONDS.sleep(5);
+
+                // Check that MQTT client has reconnected
+                logger.atInfo().log("Checking MQTT Reconnected");
+                mqttClient.waitForReconnect(configTimeout);
+
+                // Check that HTTP client works
+                logger.atInfo().log("Checking HTTP Reconnected");
+                thingGroupHelper.waitForReconnect(configTimeout);
+            } catch (Exception e) {
+                activator.mergeConfig(currentConfig, deployment);
```

Comment:
this is not the right way to rollback, so fix this by not changing the configuration.